uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveRequestTest.php

```diff
@@ -664,6 +665,128 @@ public function testALeaveRequestEndDateShouldNotBeGreaterThanStartDate() {
     ]);
   }
 
+  /**
+   * @expectedException CRM_HRLeaveAndAbsences_Exception_InvalidLeaveRequestException
+   * @expectedExceptionMessage Absence Type is not active
+   */
+  public function testLeaveRequestShouldNotBeCreatedWhenAbsenceTypeIsNotActive() {
+    $absenceType = AbsenceTypeFabricator::fabricate([
+      'is_active' => 0
+    ]);
+
+    $fromDate = new DateTime();
+    $toDate = new DateTime('+4 days');
+    LeaveRequest::create([
+      'type_id' => $absenceType->id,
+      'contact_id' => 1,
+      'status_id' => 1,
+      'from_date' => $fromDate->format('YmdHis'),
+      'from_date_type' => 1,
+      'to_date' => $toDate->format('YmdHis'),
+      'to_date_type' => 1
+    ]);
+  }
+
+  /**
+   * @expectedException CRM_HRLeaveAndAbsences_Exception_InvalidLeaveRequestException
+   * @expectedExceptionMessage Leave Request days cannot be greater than maximum consecutive days for absence type
+   */
+  public function testNumberOfDaysOfLeaveRequestShouldNotBeGreaterMaxConsecutiveLeaveDaysForAbsenceType() {
+    $absenceType = AbsenceTypeFabricator::fabricate([
+      'max_consecutive_leave_days' => 2
+    ]);
+
+    $fromDate = new DateTime();
+    $toDate = new DateTime('+4 days');
+    LeaveRequest::create([
+      'type_id' => $absenceType->id,
+      'contact_id' => 1,
+      'status_id' => 1,
+      'from_date' => $fromDate->format('YmdHis'),
+      'from_date_type' => 1,
+      'to_date' => $toDate->format('YmdHis'),
+      'to_date_type' => 1
+    ]);
+  }
+
+  /**
+   * @expectedException CRM_HRLeaveAndAbsences_Exception_InvalidLeaveRequestException
+   * @expectedExceptionMessage Absence Type does not allow leave request cancellation
+   */
+  public function testAUserCannotCancelOwnLeaveRequestWhenAbsenceTypeDoesNotAllowIt() {
+    $contactID = 5;
+    $this->registerCurrentLoggedInContactInSession($contactID);
+
+    $absenceType = AbsenceTypeFabricator::fabricate([
+      'allow_request_cancelation' => AbsenceType::REQUEST_CANCELATION_NO
+    ]);
+
+    $fromDate = new DateTime();
+    $toDate = new DateTime('+4 days');
+    $leaveRequestStatuses = array_flip(LeaveRequest::buildOptions('status_id'));
+
+    $leaveRequest = LeaveRequestFabricator::fabricateWithoutValidation([
+      'type_id' => $absenceType->id,
+      'contact_id' => $contactID,
+      'status_id' => $leaveRequestStatuses['Waiting Approval'],
+      'from_date' => $fromDate->format('YmdHis'),
+      'from_date_type' => 1,
+      'to_date' => $toDate->format('YmdHis'),
+      'to_date_type' => 1
+    ]);
+
```

Comment:
I think it would be better to put a `setExpectedException` here instead of in the docblock